### ISSUE TITLE
Form not submitting when using form validator and custom reducer

### ISSUE THREAD

**Issue by AxoInsanit** [davidkpiano/react-redux-form#1059]
### The Problem
When using a custom reducer AND a form level validator the form does not submit

### Steps to Reproduce
If you set up a custom reducer in createForms AND also the form has a form level validator, the form wont submit 

```
const store = createStore(
  combineReducers({
    // ... use createForms, which will create:
    // the model reducer at "user"
    // the forms reducer at "forms" (e.g., "forms.user")
    ...createForms({
      user: customReducerUserState,
      user2: initialUserState2
    })
  }),
  composeEnhancers()
);
```
```
    const passwordsMatch = val => {
      // console.log('val',val);
      return true;
    };
```
```
<Form
          model="user"
          onSubmit={v => console.log("submit", v)}
          validators={{
            "": { passwordsMatch }
          }}
        >
```

### Expected Behavior
The form should submit :)

### Actual Behavior
It doesnt

### Reproducible Code Example
https://codesandbox.io/s/llo0jrpyq9

**Comment by davidkpiano** [comment]
Is there any way you can avoid using Immutable?

**Comment by AxoInsanit** [comment]
Not really, no

**Comment by AxoInsanit** [comment]
is this a known issue? is there a way to make this work without getting rid of immutable?

BTW, thanks for all the great work!

**Comment by davidkpiano** [comment]
When I got rid of immutable, it worked as expected. In the next version (or a future form library that improves over `react-redux-form`) I would recommend _not_ using Immutable.JS. Working with ad-hoc data structures is a major pain point and creates unnatural, hard-baked dependencies. I recommend working with something like https://github.com/mweststrate/immer or https://github.com/aearly/icepick instead.

**Comment by AxoInsanit** [comment]
Thanks for the recommendation, sadly in this case we already have a lot of code tied to immutable as well as RRF.
is this something fixable (should I attempt a PR?, where should I start looking?) or is tied to architectural stuff ?

**Comment by davidkpiano** [comment]
> should I attempt a PR?

Yes please. PRs always welcome, and I get around to reviewing/accepting/releasing them much, much faster than fixing it myself, due to time constraints.
